lib/ukstore/store.c

```diff
@@ -29,3 +29,1194 @@
+{
+	UK_ASSERT(e);
+	switch (e->type) {
+	SETCASE_DOWNCASTSS(e, s8, S8, val, s32);
+	SETCASE_DOWNCASTSU(e, u8, U8, val, s32);
+	SETCASE_DOWNCASTSS(e, s16, S16, val, s32);
+	SETCASE_DOWNCASTSU(e, u16, U16, val, s32);
+	SETCASE_UPCAST(e, s32, S32, val, s32);
+	SETCASE_UPCASTSU(e, u32, S32, val, s32);
+	SETCASE_UPCAST(e, s64, S64, val, s32);
+	SETCASE_UPCASTSU(e, u64, U64, val, s32);
+	SETCASE_UPCASTSU(e, uptr, PTR, val, s32);
+
+	case uk_store_get_entry_type(charp): {
+		int ret;
+		char to_set[UK_STORE_NR_MAX_LEN];
+
+		sprintf(to_set, "%d", val);
+		ret = e->set.charp(to_set);
+		return ret;
+	}
+
+	default:
+		return -EINVAL;
+	}
+	return 0;
+}
+
+int
+_uk_store_set_u64(struct uk_store_entry *e, __u64 val)
+{
+	UK_ASSERT(e);
+	switch (e->type) {
+	SETCASE_DOWNCASTUS(e, s8, S8, val, u64);
+	SETCASE_DOWNCASTUU(e, u8, U8, val, u64);
+	SETCASE_DOWNCASTUS(e, s16, S16, val, u64);
+	SETCASE_DOWNCASTUU(e, u16, U16, val, u64);
+	SETCASE_DOWNCASTUS(e, s32, S32, val, u64);
+	SETCASE_DOWNCASTUU(e, u32, U32, val, u64);
+	SETCASE_DOWNCASTUS(e, s64, S64, val, u64);
+	SETCASE_UPCAST(e, u64, U64, val, u64);
+	SETCASE_UPCAST(e, uptr, PTR, val, u64);
+
+	case uk_store_get_entry_type(charp): {
+		int ret;
+		char to_set[UK_STORE_NR_MAX_LEN];
+
+		sprintf(to_set, "%lu", val);
+		ret = e->set.charp(to_set);
+		return ret;
+	}
+
+	default:
+		return -EINVAL;
+	}
+	return 0;
+}
+
+int
+_uk_store_set_s64(struct uk_store_entry *e, __s64 val)
+{
+	UK_ASSERT(e);
+	switch (e->type) {
+	SETCASE_DOWNCASTSS(e, s8, S8, val, s64);
+	SETCASE_DOWNCASTSU(e, u8, U8, val, s64);
+	SETCASE_DOWNCASTSS(e, s16, S16, val, s64);
+	SETCASE_DOWNCASTSU(e, u16, U16, val, s64);
+	SETCASE_DOWNCASTSS(e, s32, S32, val, s64);
+	SETCASE_DOWNCASTSU(e, u32, S32, val, s64);
+	SETCASE_UPCAST(e, s64, S64, val, s64);
+	SETCASE_UPCASTSU(e, u64, U64, val, s64);
+	SETCASE_UPCASTSU(e, uptr, PTR, val, s64);
+
+	case uk_store_get_entry_type(charp): {
+		int ret;
+		char to_set[UK_STORE_NR_MAX_LEN];
+
+		sprintf(to_set, "%ld", val);
+		ret = e->set.charp(to_set);
+		return ret;
+	}
+
+	default:
+		return -EINVAL;
+	}
+	return 0;
+}
+
+int
+_uk_store_set_uptr(struct uk_store_entry *e, __uptr val)
+{
+	UK_ASSERT(e);
+	switch (e->type) {
+	SETCASE_DOWNCASTUS(e, s8, S8, val, uptr);
+	SETCASE_DOWNCASTUU(e, u8, U8, val, uptr);
+	SETCASE_DOWNCASTUS(e, s16, S16, val, uptr);
+	SETCASE_DOWNCASTUU(e, u16, U16, val, uptr);
+	SETCASE_DOWNCASTUS(e, s32, S32, val, uptr);
+	SETCASE_DOWNCASTUU(e, u32, U32, val, uptr);
+	SETCASE_DOWNCASTUS(e, s64, S64, val, uptr);
+	SETCASE_UPCAST(e, u64, U64, val, uptr);
+	SETCASE_UPCAST(e, uptr, PTR, val, uptr);
+
+	case uk_store_get_entry_type(charp): {
+		int ret;
+		char to_set[UK_STORE_NR_MAX_LEN];
+
+		sprintf(to_set, "%p", (void *) val);
+		ret = e->set.charp(to_set);
+		return ret;
+	}
+
+	default:
+		return -EINVAL;
+	}
+	return 0;
+}
+
+int
+_uk_store_set_charp(struct uk_store_entry *e, char *val)
+{
+	return _uk_store_set_s64(e, atoi(val));
```

Comment:
Oh, I think this could be to easy. u64 would not be fully covered. Can you try to use `sscanf` and the appropriate `__SCN*` format definitions?